relay-event-normalization/src/transactions/processor.rs

```diff
@@ -163,14 +179,61 @@ impl Processor for TransactionsProcessor<'_> {
         _meta: &mut Meta,
         state: &ProcessingState<'_>,
     ) -> ProcessingResult {
-        span.op.get_or_insert_with(|| "default".to_owned());
-
+        if span.op.value().is_none() {
+            *span.op.value_mut() = Some(self.span_op_defaults.infer(span));
+        }
         span.process_child_values(self, state)?;
 
         Ok(())
     }
 }
 
+/// Rules used to infer `span.op` from other span fields.
+#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq)]
+pub struct SpanOpDefaults {
+    /// List of rules to apply. First match wins.
+    pub rules: Vec<SpanOpDefaultRule>,
+}
+
+impl SpanOpDefaults {
+    /// Gets a borrowed version of this config.
+    pub fn borrow(&self) -> BorrowedSpanOpDefaults {
+        BorrowedSpanOpDefaults {
+            rules: self.rules.as_slice(),
+        }
+    }
+}
```

Comment:
I'd impl `AsRef` here instead

Reply:
Hmm, how would I, actually? `AsRef` and `std::borrow::Borrow` both return a reference to its type, but I want to return a new, reference-like struct.

Reply:
Sorry, that's not actually possible.